pkg/filebacked/file.go

```diff
@@ -76,25 +82,32 @@ func (w *Writer) Append(object interface{}) {
 }
 
 //
-// Get a reader.
-func (w *Writer) Reader() (reader *Reader) {
-	err := w.file.Sync()
-	if err != nil {
-		panic(err)
-	}
-	path := w.newPath()
-	err = os.Link(w.path, path)
-	if err != nil {
-		panic(err)
-	}
-	reader = &Reader{
-		path: path,
+// Build a reader.
+func (w *Writer) Reader(shared bool) (reader *Reader) {
```

Comment:
Add concept of _shared_ reader to efficiently support both read and write operations.  Besides not linking the file we don't want to set the finalizer either.